zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -198,8 +198,13 @@ private boolean isUserAuthorizedForTenant(
     if (user.getTenantIdsList().contains(tenantId)) {
       return true;
     }
+    final var groupIds =
+        membershipState.getMemberships(
+            EntityType.USER, Long.toString(user.getUserKey()), RelationType.GROUP);
+    // TODO: use group id instead of converting to a key when groups are id-based
+    final var groupKeys = groupIds.stream().map(Long::parseLong).collect(Collectors.toSet());
```

Comment:
🔧 You have conflicts in the PR, likely because this refactoring you refer to in this TODO has been done already. So ideally you just use the username to get the memberships here and work with group ids instead.